js/search-form.js

```diff
@@ -0,0 +1,83 @@
+const spinner = document.getElementById("spinner");
+
+function debounce(func, interval) {
+  var timeout;
+  return function () {
+    var context = this,
+      args = arguments;
+    var later = function () {
+      timeout = null;
+      func.apply(context, args);
+    };
+    clearTimeout(timeout);
+    timeout = setTimeout(later, interval || 200);
+  };
+}
+
+class SearchForm {
+  constructor(formElement) {
+    this.formElement = formElement;
+    this.input = this.formElement.querySelector("#user-input");
+    this.setListener();
+  }
+
+  setListener() {
+    this.input.addEventListener(
+      "input",
+      debounce((event) => {
+        event.preventDefault();
+        document.getElementById("results").innerHTML = "";
+        spinner.classList.remove("transparent");
+        this.getSearchData();
+      }, 1000)
```

Comment:
try shorter deboucing time for better user experience. the convention is around 300-400 miliseconds.